python_modules/libraries/dagstermill/dagstermill/manager.py

```diff
@@ -335,13 +335,21 @@ def yield_result(self, value, output_name="result"):
         # pass output value cross process boundary using io manager
         step_context = self.context._step_context  # pylint: disable=protected-access
         # Note: yield_result currently does not support DynamicOutput
+
+        # dagstermill assets do not support yielding additional results within the notebook:
+        if len(step_context.job_def.asset_layer.asset_keys) > 0:
+            raise DagstermillError(
+                "dagstermill assets do not currently support dagstermill.yield_result"
+            )
+
         step_output_handle = StepOutputHandle(
             step_key=step_context.step.key, output_name=output_name
         )
         output_context = step_context.get_output_context(step_output_handle)
         io_manager = step_context.get_io_manager(step_output_handle)
 
         # Note that we assume io manager is symmetric, i.e handle_input(handle_output(X)) == X
+        # use list() in case handle_output is a generator
```

Comment:
did you mean to use `list` here?

Reply:
forgot to delete the comment, oops. thanks for the catch